src/main/java/net/countercraft/movecraft/combat/features/tracking/BlockBehaviorOverride.java

```diff
@@ -0,0 +1,420 @@
+            try {
+                final Object block = getBlockClass(m);
+                // First Object2Identiy map in field list is the one for ignite odds, second one is for burn odds
+                Consumer<Object2IntMap> func = (map) -> {
+                    map.put(block, value);
+                };
+                NMSSpigotMappings.writeField(fireBlock, func, this.fieldNameIgniteOdds);
+                return true;
+            } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException
+                     | NoSuchMethodException | NoSuchFieldException
+                     | SecurityException | ClassNotFoundException e) {
+                e.printStackTrace();
+                return false;
+            }
+        }
+
+        abstract Class<?> getCraftMagicNumbersClass() throws ClassNotFoundException;
+
+        protected Object getBlockClass(Material m)
+                throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, ClassNotFoundException {
+            if (this.magicNumbers == null) {
+                this.magicNumbers = this.getCraftMagicNumbersClass();
+            }
+            Method method = magicNumbers.getMethod("getBlock", Material.class);
+            return method.invoke(null, m);
+        }
+
+        protected static <T> void writeField(@NotNull Object block, @NotNull Consumer<T> whatToDoWithField, String fieldName) throws IllegalAccessException, NoSuchFieldException, ClassCastException {
+            Field field = FieldUtils.getField(block.getClass(), fieldName, true);
+            T obj = (T)field.get(block);
+            whatToDoWithField.accept(obj);
+        }
+
+        protected static <T> void writeField(@NotNull Object block, T value, String fieldName) throws IllegalAccessException, NoSuchFieldException, ClassCastException {
+            Field field = FieldUtils.getField(block.getClass(), fieldName, true);
+            T obj = (T)field.get(block);
+            field.set(block, value);
+        }
+
+        protected static <T> Optional<T> getFieldValueSafe(@NotNull Object instance, String fieldName) {
+            try {
+                return Optional.ofNullable(getFieldValue(instance, fieldName));
+            } catch(Exception ex) {
+                return Optional.empty();
+            }
+        }
+        protected static <T> T getFieldValue(@NotNull Object instance, String fieldName) throws IllegalAccessException, NoSuchFieldException, ClassCastException {
+            Field field = FieldUtils.getField(instance.getClass(), fieldName, true);
+            T obj = (T)field.get(instance);
+            return obj;
+        }
+    }
+
+    private static class NMSSpigotMappings extends NMSHelper {
+
+        private static final String FIELD_NAME_IGNITE_ODDS = "O";
+        private static final String FIELD_NAME_BURN_ODDS = "P";
+        private static final String FIELD_NAME_BLAST_RESISTANCE = "aH";
```

Comment:
Has this been tested on 1.18.2?  If so we should have a comment for it.

Reply:
Didn't test it on 1.18.2, but i'm pretty sure the field names are the same there, the "aH" fieldname is directly taken from the old blast resistance override version